clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -374,8 +388,48 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         }
         // Invariant: non-explicit control flow (i.e., kernel-mediated) is indicated
         // by markers.
-        const std::string non_explicit_flow_violation_msg =
-            check_for_pc_discontinuity(shard, memref);
+        bool have_cond_branch_target = false;
```

Comment:
Both the branch target check and the repeated syscall check should be done inside check_for_pc_discontinuity, since they are also checks for pc discontinuities.

Reply:
Done. Moved them inside the function and passed `cur_instr_decoded` and `expect_encoding` to it.